### DISCUSSION TITLE
`StreamGeometryContext` or `StreamGeometry` in WinUI 3 ?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9037] [Q&A]
_[From a discord conversation:](https://discord.com/channels/372137812037730304/550416151172087808/985890014556725318)_

Hi, is there a way to use StreamGeometryContext or StreamGeometry (WPF stuff) in WinUI 3 ?

**Comment by jeromelaban** [comment]
Could you provide an example of what you're trying to do?

**Comment by x86** [comment]
I am trying to port https://github.com/microsoft/automatic-graph-layout/ to Uno.
I made the AutomaticGraphLayout and AutomaticGraphLayout.Drawing .NET 6 compatible (just by adding .NET6 into TargetFrameworks). After that I copied Tools/UwpGraphControl to Tools/UnoGraphControl and started to convert this from UWP to WinUI 3 except the actual drawing of those things (arrows, lines, ellipse and so on). The StreamGeometry and StreamGeometryContext classes are being used in VEdge.cs.

**Reply by jeromelaban** [reply]
In the context of this library, `StreamGeometry` is imported from the WPF namespace, making it incompatible with WinUI, and certainly not available cross platform.

To port this code, you'll likely need to port the `StreamGeometry` class from WPF as well.

Note that Uno [exposes its own version of `StreamGeometry`](https://github.com/unoplatform/uno/blob/833a6e1f358041f4fb61dc1c0ffd2c8b7519674a/src/Uno.UI/Media/StreamGeometry.cs#L39), but since it's not a WinUI API, it may change in the future.

**Reply by x86** [reply]
Is there any alternative (library) to achieve this?

**Reply by lindexi** [reply]
@jeromelaban Two years later, are you considering public this type now?

**Reply by jeromelaban** [reply]
I do not think that WinUI has changed on that side at this time, so this a status quo at this point.

**Reply by lindexi** [reply]
@jeromelaban When will you consider expanding UNO's API set beyond that of WinUI? I'm concerned that the lack of development in WinUI could potentially undermine UNO's competitiveness.

**Reply by jeromelaban** [reply]
We're already doing this on other parts, in toolkit or others. So far, we intend to keep alignments with WinUI where it makes sense.

**Reply by lindexi** [reply]
Looking forward to your good news.
